src/components/dashboard/Dashboard.js

```diff
@@ -148,6 +150,14 @@ const Dashboard = props => {
     setAvatarCenteredPosition(newAvatarCenteredPosition)
   }, [setHeaderContentWidth, setAvatarCenteredPosition])
 
+  const balanceFormatter = useMemo(
+    () =>
+      headerLarge || Math.floor(Math.log10(balance)) + 1 <= 12
+        ? null
+        : _balance => formatWithAbbreviations(weiToGd(_balance), 2),
```

Comment:
1. declare `const abbreviateBalance = value =>  formatWithAbbreviations(weiToGd(_balance), 2)`
2. use `abbreviateBalance` here

this will optimize your memo (no new instance will be produced on each balance / headerLarge change)